### DISCUSSION TITLE
Hi everyone! i have a following question: how to set a timeout to soap call???

### DISCUSSION THREAD

**Discussion by sramirezse** [CodeDredd/laravel-soap#209] [Q&A] [answered]
I tried to use withOptions(['connection_timeout' and 'default_socket_timeout' but the response takes much longer and does not throw an error, this is my code:

`try{

            $responseGetOrder = Soap::baseWsdl($this->url)->withOptions([
                'connection_timeout' => 30,
                'default_socket_timeout' => 30,
            ])->getOrder([
                'user' => $this->user,
                'operator' => $this->operator,
                'idProduct' => 'RMP100',
                'numberAccount' => 5511111160,
                // 'numberAccount' => 3334441122,
                // 'numberAccount' => 5511111130,
                'clientFolio' => $serviceAttempt->id,
            ]);
            $responseGetOrderBody = $responseGetOrder->json();
            return $responseGetOrderBody;
        }catch(\Exception $e){
           dd($e);
        }`
this is the time it takes to respond, sorry my bad english
![imagen](https://user-images.githubusercontent.com/49218071/159870837-eb4d23c3-4666-44a0-88a1-98048a2fb178.png)

**Comment by CodeDredd** [comment, marked as answer]
Answered in issue #210
